File: bindings/typescript/examples/streams.mjs
// Print three public Binance trade-stream items, then close the iterator.
import {
  BinanceAdapter,
  Client,
  Exchange,
  Feed,
  Market,
  Subscription,
  initialize,
} from "@jabdori/maxt/node";

await initialize();

const market = Market.spot(Exchange.Binance, "BTC", "USDT");
const stream = await new Client(BinanceAdapter.spot()).subscribe(
  new Subscription([market], [Feed.Trades]),
);

let remaining = 3;
for await (const item of stream) {
  console.log(item);
  if (--remaining === 0) break;
}

File: bindings/dart/example/streams.dart
import 'package:maxt/maxt.dart';

Future<void> main() async {
  await Maxt.initialize();
  try {
    final market = Market.spot(Exchange.binance, 'BTC', 'USDT');
    final client = Client(BinanceAdapter.spot());
    final stream = await client.subscribe(
      Subscription(markets: [market], feeds: [Feed.trades]),
    );

    var remaining = 3;
    await for (final item in stream) {
      print(item);
      if (--remaining == 0) break;
    }
  } finally {
    await Maxt.dispose();
  }
}
